# Flipkart Mobile Dataset

In [1]:
import pandas as pd

In [2]:
dataset = pd.read_csv("Flipkart Mobile.csv")

In [3]:
dataset

brand           model base_color processor screen_size  ROM  RAM  \
0     Apple       iPhone SE      Black     Water  Very Small   64    2   
1     Apple  iPhone 12 Mini        Red   Ceramic       Small   64    4   
2     Apple       iPhone SE        Red     Water  Very Small   64    2   
3     Apple       iPhone XR     Others       iOS      Medium   64    3   
4     Apple       iPhone 12        Red   Ceramic      Medium  128    4   
..      ...             ...        ...       ...         ...  ...  ...   
425  Xiaomi     Redmi 6 Pro      Black  Qualcomm       Small   32    3   
426  Xiaomi     Redmi 6 Pro        Red  Qualcomm       Small   64    4   
427  Xiaomi      Mi 11 Lite     Others  Qualcomm       Large  128    6   
428  Xiaomi   Redmi 8A Dual       Blue  Qualcomm      Medium   32    3   
429  Xiaomi     Redmi 6 Pro       Blue  Qualcomm       Small   32    3   

     display_size  num_rear_camera  num_front_camera  battery_capacity  \
0             4.7                1                 1              1800   
1             5.4                2                 1              2815   
2             4.7                1                 1              1800   
3             6.1                1                 1              2942   
4             6.1                2                 1              2815   
..            ...              ...               ...               ...   
425           5.8                2                 1              4000   
426           5.8                2                 1              4000   
427           6.5                3                 1              4250   
428           6.2                2                 1              5000   
429           5.8                2                 1              4000   

     ratings  num_of_ratings  sales_price  discount_percent   sales  
0        4.5           38645        32999              0.17  127.52  
1        4.5             244        57149              0.04    1.39  
2        4.5           38645        32999              0.17  127.52  
3        4.6            5366        42999              0.10   23.07  
4        4.6             745        69149              0.02    5.15  
..       ...             ...          ...               ...     ...  
425      4.3            1870         7999              0.30    1.50  
426      4.3            1783         9699              0.28    1.73  
427      4.2            1554        21999              0.12    3.42  
428      4.2            8161         8299              0.07    6.77  
429      4.3            1870         8190              0.36    1.53  

[430 rows x 16 columns]

# DATASET Description

ABOUT DATASET

The dataset includes data on mobile phones from the top five most popular 
brands in India: Apple, Poco, Realme, Samsung, and Xiaomi. Information like 
RAM, ROM, Display Size. etc are present which distinguishes one product from 
another. At least one attribute distinguishes each product. Dataset has no null 
value.

Columns: There are 16 columns each having a title which is self-explanatory.
Rows: There are 430 rows each having a mobile with at least a distinct feature.

DESCRIPTION OF ATTRIBUTES

I. brand: Brand Name (Categorical)

II. model: Model Name (Categorical)

III. base_color: Phone Color (Categorical)

IV. processor: Processor brand used (Categorical)

V. screen_size: Categorical screen size (Categorical)

VI. ROM: ROM in gigabyte (Numeric – Discrete)

VII. RAM: RAM in gigabyte (Numeric – Discrete)

VIII. display_size: Actual display size in inches (Numeric – Continuous)

IX. num_rear_camera: No. of cameras on back (Numeric – Discrete)

X. num_front_camera: No. of cameras on front (Numeric – Discrete)

XI. battery_size: Battery in mAH (Numeric – Continuous)

XII. ratings: Customer rating for the product (Numeric – Continuous)

XIII. num_of_ratings: No. of people rating the product, also the equivalent 
no. of unit sold for our problem (Numeric – Continuous)

XIV. sales_price: Selling price of the unit after discount (Numeric –
Continuous)

XV. discount_percent: Discount in percentage offered (Numeric –
Continuous)

XVI. sales: Sales of product in crore rupees (Numeric – Continuous)

